### ISSUE TITLE
Provide API for IAssemblySymbol to retrieve type forwards

### ISSUE THREAD

**Issue by agocke** [dotnet/roslyn#36274]
There's already an internal API on PEModuleSymbol, GetForwardedTypes, but I don't think it's ready to be made public. We should actually do some design work to figure out how we want this API to behave.

**Comment by safern** [comment]
It would be really nice to have this. When iterating members from an assembly symbol gotten via a `CSharpCompilation` object where you need to visit all types including forwarded ones, it is not convenient to have an use the MetadataReader to get the list of forwarded types and then call `IAssemblySymbol.ResolveForwardedType` for every forwarded type.

**Comment by agocke** [comment]
fyi @jaredpar for scheduling

**Comment by mgoertz-msft** [comment]
Any idea when we can get this API? The issue keeps coming up every so often and is rather confusing to the user: https://github.com/dotnet/wpf/issues/2441

**Comment by vatsan-madhavan** [comment]
"clr-namespace:System;assembly=mscorlib"` is a common pattern. Issues like https://github.com/dotnet/wpf/issues/2441 seem to indicate that mscorlib + XAML doesn't work as well as we had hoped. 

This is a source of friction for migration of WPF applications from .NET framework to .NET Core. For the vast majority of applications, our goal has been to enable relatively pain-free migration. To that end, we've maintained a very high degree of API compatibility between .NET Framework 4.8 and .NET Core 3.1 (in fact, internally within the team, we expect those two products to work functionally identically, modulo a small number of features that were excised from the .NET Core codebase). 

In WPF, this essentially means the ability to consume API's either in procedural code-behind, or in XAML. Asking customers to edit XAML is not going to work well in practice and will be prone to errors or resistance; I know this from experience as I spent a lot of time doing exactly this sort of XAML-fixup for a subset of WPF's test codebase early in the prototyping stage, when WPF's markup compiler still had some problems and we needed to bootstrap tests anyway.
